internal/search/run/repository.go

```diff
@@ -53,7 +53,7 @@ func (s *RepoSearch) Run(ctx context.Context, db database.DB, stream streaming.S
 		return nil
 	})
 
-	if errors.Is(err, searchrepos.ErrNoResolvedRepos) && len(s.Args.RepoOptions.Dependencies) == 0 {
+	if errors.Is(err, searchrepos.ErrNoResolvedRepos) && s.Args.PatternInfo.Pattern != "" {
 		err = nil
 	}
```

Comment:
aha! This is what I was missing. 

nit: this might not matter right now, but this will also ignore errors wrapped with a `ErrNoResolvedRepos` in a multierror. something like this might save some future headaches:

```suggestion
if s.Args.PatternInfo.Pattern != "" {
    err = errors.Ignore(err, errors.IsPred(searchrepos.ErrNoResolvedRepos))
}
```

Reply:
This feels like the right pattern moving forward: jobs handle and convert errors, then the alert code translates errors into alerts.